src/enzo/Grid_DepositBaryons.C

```diff
@@ -235,7 +237,25 @@ int grid::DepositBaryons(grid *TargetGrid, FLOAT DepositTime)
 	av_dens = BaryonField[DensNum];
 
       input_density = av_dens;
```

Comment:
Since `input_density` is now being allocated with `new` this assignment is bad. This should be changed to: `for (int i = 0; i < current_size; i++) input_density[i] = av_dens[i];`

Reply:
fixed